### PR TITLE
feat(nagpack): foundations for Prototyping NagPack

### PR DESCRIPTION
**Pull request by dboyd13**
Foundations for a PDK custom NagPack, including

- Pack and rule meta definition
- AwsPrototypingChecks NagPack (based off of pack and rule meta definition)
- Build-time generation of pack documentation (based off of pack and rule meta definition)

### CODE REVIEW

File: packages/pdk-nag/scripts/generate-pack-markdown.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+SPDX-License-Identifier: Apache-2.0 */
+import { NagMessageLevel } from "cdk-nag";
+import * as fs from "fs-extra";
+import * as Mustache from "mustache";
+import { PackName, RuleMetadata } from "../src/packs/aws-prototyping-rules";
+
+const renderedMarkdown = Mustache.render(
+  fs
+    .readFileSync(
+      `${__dirname}/templates/markdown/awsprototypingrules-readme.mustache`
```
Comment (agdimech) [inline packages/pdk-nag/scripts/generate-pack-markdown.ts:11]: Can you please confirm the mustache file is present in `dist/js/pdk-nag@0.0.0.jsii.tgz`

File: packages/pdk-nag/src/packs/aws-prototyping.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+/**
+ * Check Best practices for prototypes
+ *
+ */
+export class AwsPrototypingChecks extends NagPack {
+  constructor(props?: NagPackProps) {
+    super(props);
+    this.packName = PackName;
+  }
+  public visit(node: IConstruct): void {
```
Comment (agdimech) [inline packages/pdk-nag/src/packs/aws-prototyping.ts:21]: missing jsdocs

File: packages/pdk-nag/src/packs/aws-prototyping-rules.ts [hunk 3]
```diff
@@ -0,0 +1,29 @@
+/*! Copyright [Amazon.com](http://amazon.com/), Inc. or its affiliates. All Rights Reserved.
+SPDX-License-Identifier: Apache-2.0 */
+import { NagMessageLevel, rules } from "cdk-nag";
+
+export let PackName = "AwsPrototyping";
+
+export let RuleMetadata = [
```
Comment (agdimech) [inline packages/pdk-nag/src/packs/aws-prototyping-rules.ts:15]: missing jsdocs

File: packages/pdk-nag/src/packs/aws-prototyping-rules.ts [hunk 4]
```diff
@@ -0,0 +1,29 @@
+/*! Copyright [Amazon.com](http://amazon.com/), Inc. or its affiliates. All Rights Reserved.
+SPDX-License-Identifier: Apache-2.0 */
+import { NagMessageLevel, rules } from "cdk-nag";
+
+export let PackName = "AwsPrototyping";
```
Comment (agdimech) [inline packages/pdk-nag/src/packs/aws-prototyping-rules.ts:9]: missing jsdocs

File: packages/pdk-nag/src/packs/aws-prototyping.ts [hunk 5]
```diff
@@ -0,0 +1,28 @@
+/*! Copyright [Amazon.com](http://amazon.com/), Inc. or its affiliates. All Rights Reserved.
+SPDX-License-Identifier: Apache-2.0 */
+
+import { CfnResource } from "aws-cdk-lib";
+import { NagPack, NagPackProps } from "cdk-nag";
+import { IConstruct } from "constructs";
+import { PackName, RuleMetadata } from "./aws-prototyping-rules";
+
+/**
+ * Check Best practices for prototypes
```
Comment (agdimech) [inline packages/pdk-nag/src/packs/aws-prototyping.ts]: best*

**Review by agdimech — COMMENTED** [review]
Looking good, added minor comments.

File: packages/pdk-nag/scripts/generate-pack-markdown.ts (hunk 1 above)
Comment (dboyd13) [reply packages/pdk-nag/scripts/generate-pack-markdown.ts:11]: Within `pdk-nag@0.0.0.jsii.tgz` the `awsprototypingrules-readme.mustache` file is present in `/packages/pdk-nag/dist/js/package/scripts/templates/markdown`

File: packages/pdk-nag/src/packs/aws-prototyping.ts (hunk 5 above)
Comment (dboyd13) [reply packages/pdk-nag/src/packs/aws-prototyping.ts]: Fixed in [43523519d4a565d9d12cb76e4e1335a25fcea7b2](https://github.com/aws/aws-prototyping-sdk/pull/213/commits/43523519d4a565d9d12cb76e4e1335a25fcea7b2)

File: packages/pdk-nag/src/packs/aws-prototyping-rules.ts (hunk 4 above)
Comment (dboyd13) [reply packages/pdk-nag/src/packs/aws-prototyping-rules.ts:9]: Fixed in [43523519d4a565d9d12cb76e4e1335a25fcea7b2](https://github.com/aws/aws-prototyping-sdk/pull/213/commits/43523519d4a565d9d12cb76e4e1335a25fcea7b2)

File: packages/pdk-nag/src/packs/aws-prototyping-rules.ts (hunk 3 above)
Comment (dboyd13) [reply packages/pdk-nag/src/packs/aws-prototyping-rules.ts:15]: Fixed in [43523519d4a565d9d12cb76e4e1335a25fcea7b2](https://github.com/aws/aws-prototyping-sdk/pull/213/commits/43523519d4a565d9d12cb76e4e1335a25fcea7b2)

File: packages/pdk-nag/src/packs/aws-prototyping.ts (hunk 2 above)
Comment (dboyd13) [reply packages/pdk-nag/src/packs/aws-prototyping.ts:21]: Fixed in [43523519d4a565d9d12cb76e4e1335a25fcea7b2](https://github.com/aws/aws-prototyping-sdk/pull/213/commits/43523519d4a565d9d12cb76e4e1335a25fcea7b2)

**Review by agdimech — DISMISSED** [review]

**Review by agdimech — APPROVED** [review]

**Pull request merged**